### ISSUE TITLE
Block while typing after #1583

### ISSUE THREAD

**Issue by fecet** [hrsh7th/nvim-cmp#1606]
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt) and it didn't resolve my problem.

### Announcement

- [X] I have checked [Breaking change announcement](https://github.com/hrsh7th/nvim-cmp/issues/231).

### Minimal reproducible full config

```vim

```

### Description

I found completion become very slow after update, a git bisect tell me the first error happen at abb5c7519d40314240effc70d781149c0f097af3.

https://github.com/hrsh7th/nvim-cmp/assets/41792945/0758a705-9cd0-493d-8199-bb20627f0e8b

### Steps to reproduce

1. Open any python file
2. type a function name, like "print" or "isinstance"
3. type anything, it will block at first character

### Expected behavior

cmp will not block my typing

### Actual behavior

see above

**Comment by folke** [comment]
What is needed to reproduce this? When I open a python file and start typing and don't experience any delays.

What other lsp completion related plugins are you using?

**Comment by fecet** [comment]
Perhaps relavent to https://github.com/ray-x/lsp_signature.nvim and I'm using pylsp

**Comment by folke** [comment]
Tried that, but still doesn't lag.

**Comment by folke** [comment]
What is your lsp signature and pylsp config?

**Comment by folke** [comment]
Can you give me a single python source file that triggers the issue with the exact steps to trigger it?

**Comment by fecet** [comment]
Do you want a minimun config or just reproduce, is $NVIM_APPNAME ok?

**Comment by folke** [comment]
minimum config, since I need to be able to properly reproduce this.

You can use this template https://github.com/folke/noice.nvim/wiki/Minimal-%60init.lua%60-to-Reproduce-an-Issue

**Comment by folke** [comment]
Not sure it will fix anything, but if you could try https://github.com/hrsh7th/nvim-cmp/pull/1611 to see if that makes a difference.

There was an issue where a cancelled/stopped filter would not set `running = false`.

**Comment by fecet** [comment]
I tried to extract all my relavent config but unfortunately the bug doesn't happen...It seems that someone have provided a config and you have found the reasons behind the bug, tell me if you still need a minimal config for python.

 I hope this can help you eliminate some problems:
```lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
	vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		"git",
		"clone",
		"--filter=blob:none",
		"--single-branch",
		"https://github.com/folke/lazy.nvim.git",
		lazypath,
	})
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
	"folke/tokyonight.nvim",
	{
		"neovim/nvim-lspconfig",

		lazy = true,
		event = { "BufReadPost", "BufAdd", "BufNewFile" },
		config = function()
			local nvim_lsp = require("lspconfig")
			local mason = require("mason")
			local mason_registry = require("mason-registry")
			local mason_lspconfig = require("mason-lspconfig")
			require("lspconfig.ui.windows").default_options.border = "single"

			mason.setup({
				ui = {
					border = "single",
					keymaps = {
						toggle_server_expand = "<CR>",
						install_server = "i",
						update_server = "u",
						check_server_version = "c",
						update_all_servers = "U",
						check_outdated_servers = "C",
						uninstall_server = "X",
						cancel_installation = "<C-c>",
					},
				},
			})
			mason_registry:on(
				"package:install:success",
				vim.schedule_wrap(function(pkg)
					if pkg.name ~= "python-lsp-server" then
						return
					end

					local venv = vim.fn.stdpath("data") .. "/mason/packages/python-lsp-server/venv"
					require("plenary.job")
						:new({
							command = venv .. "/bin/python",
							args = {
								"-m",
								"pip",
								"install",
								"-U",
								"--disable-pip-version-check",
								"python-lsp-black",
								"python-lsp-ruff",
								"pylsp-rope",
							},
							cwd = venv,
							env = { VIRTUAL_ENV = venv },
							on_exit = function()
								if
									vim.fn.executable(venv .. "/bin/black") == 1
									and vim.fn.executable(venv .. "/bin/ruff") == 1
								then
									vim.notify(
										"Finished installing pylsp plugins",
										vim.log.levels.INFO,
										{ title = "[lsp] Install Status" }
									)
									return
								end
							end,
							on_start = function()
								vim.notify(
									"Now installing pylsp plugins...",
									vim.log.levels.INFO,
									{ title = "[lsp] Install Status", timeout = 6000 }
								)
							end,
							on_stderr = function(_, msg_stream)
								vim.notify(msg_stream, vim.log.levels.ERROR, { title = "[lsp] Install Failure" })
							end,
						})
						:start()
				end)
			)
			-- Additional plugins for pylsp
			mason_lspconfig.setup({
				ensure_installed = { "pylsp" },
			})

			local capabilities = vim.lsp.protocol.make_client_capabilities()
			capabilities = require("cmp_nvim_lsp").default_capabilities(capabilities)

			local opts = {
				on_attach = function()
					require("lsp_signature").on_attach({
						bind = true,
						use_lspsaga = false,
						floating_window = true,
						fix_pos = true,
						hint_enable = true,
						hi_parameter = "Search",
						handler_opts = {
							border = "single",
						},
					})
				end,
				capabilities = capabilities,
			}

			---A handler to setup all servers defined under `completion/servers/*.lua`
			---@param lsp_name string
			local function mason_handler(lsp_name)
				local ok, custom_handler = pcall(require, "completion.servers." .. lsp_name)
				if not ok then
					-- Default to use factory config for server(s) that doesn't include a spec
					nvim_lsp[lsp_name].setup(opts)
					return
				elseif type(custom_handler) == "function" then
					--- Case where language server requires its own setup
					--- Make sure to call require("lspconfig")[lsp_name].setup() in the function
					--- See `clangd.lua` for example.
					custom_handler(opts)
				elseif type(custom_handler) == "table" then
					nvim_lsp[lsp_name].setup(vim.tbl_deep_extend("force", opts, custom_handler))
				else
					vim.notify(
						string.format(
							"Failed to setup [%s].\n\nServer definition under `completion/servers` must return\neither a fun(opts) or a table (got '%s' instead)",
							lsp_name,
							type(custom_handler)
						),
						vim.log.levels.ERROR,
						{ title = "nvim-lspconfig" }
					)
				end
			end

			mason_lspconfig.setup_handlers({ mason_handler })
		end,
		dependencies = {
			{ "ray-x/lsp_signature.nvim" },
			{ "williamboman/mason.nvim" },
			{ "williamboman/mason-lspconfig.nvim" },
		},
	},
	{
		"m4xshen/autoclose.nvim",
		lazy = true,
		event = "InsertEnter",
		config = function()
			require("autoclose").setup({
				keys = {
					["("] = { escape = false, close = true, pair = "()" },
					["["] = { escape = false, close = true, pair = "[]" },
					["{"] = { escape = false, close = true, pair = "{}" },

					[">"] = { escape = true, close = false, pair = "<>" },
					[")"] = { escape = true, close = false, pair = "()" },
					["]"] = { escape = true, close = false, pair = "[]" },
					["}"] = { escape = true, close = false, pair = "{}" },

					['"'] = { escape = true, close = true, pair = '""' },
					["'"] = { escape = true, close = true, pair = "''" },
					["`"] = { escape = true, close = true, pair = "``" },
				},
				options = {
					disabled_filetypes = { "big_file_disabled_ft" },
					disable_when_touch = false,
				},
			})
		end,
	},

	{
		"hrsh7th/nvim-cmp",
		lazy = true,
		event = "InsertEnter",
		dependencies = {
			{ "hrsh7th/cmp-nvim-lsp" },
		},
		config = function()
			local cmp = require("cmp")
			cmp.setup({
				sources = {
					{ name = "nvim_lsp" },
				},
				window = {},

				experimental = {
					ghost_text = {
						hl_group = "Whitespace",
					},
				},
			})
		end,
	},
}
require("lazy").setup(plugins, {
	root = root .. "/plugins",
})

-- add anything else here
vim.opt.termguicolors = true
vim.cmd([[colorscheme tokyonight]])

```

**Comment by folke** [comment]
Nope, I didnt. Still don't have a way to reproduce this.

**Comment by folke** [comment]
Just created a proper fix in #1611
Would be great if you could confirm.

**Comment by fecet** [comment]
Can confirm this works!
